history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval int64) *History {
+	h := new(History)
+	h.tree = m
+	h.epochInterval = epInterval
+	h.currentSTR = newSTR(m, startEp, 0, make([]byte, crypto.HashSizeByte), key)
+	return h
+}
+
+func (h *History) UpdateHistory(m *MerkleTree, nextEp int64) error {
```

Comment:
> At t == 0, we don't have any STRs.

I don't think so. As illustration in the paper (figure 2), there is a STR for `t==0`. In the implementation, we also take a STR in the beginning (https://github.com/coniks-sys/libmerkleprefixtree-go/blob/0f821615420293b4c098ea9ccf93450950006c3d/history.go#L22), so does the reference implementation (https://github.com/coniks-sys/coniks-ref-implementation/blob/master/coniks_server/src/org/coniks/coniks_server/ServerHistory.java#L68)

> Note, I may be wrong in saying that lookups fallback to searching the temporary bindings

I think you are right at this point. @masomel also mentioned it in the temporary binding email thread.